### PR TITLE
feature/core/add-jwt-repo

### PR DESCRIPTION
**Pull request by TalhaAli00**

### CODE REVIEW

File: foundation/src/main/kotlin/com/walletconnect/foundation/di/ScarletModule.kt [hunk 1]
```diff
@@ -52,5 +52,5 @@ fun scarletModule(serverUrl: String, jwt: String): Module = module {
     }
     single { get<Scarlet>().create(RelayService::class.java) }
 
-    single<RelayInterface> { object: BaseRelayClient(get(), get(), scope) {} }
+    factory<RelayInterface> { object: BaseRelayClient(get(), get(), scope) {} }
```
Comment (Elyniss) [inline foundation/src/main/kotlin/com/walletconnect/foundation/di/ScarletModule.kt:55]: Why factory?

File: foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        data[i / 2] = ((Character.digit(this[i], 16) shl 4)
+                + Character.digit(this[i + 1], 16)).toByte()
+        i += 2
+    }
+
+    return data
+}
+
+@JvmSynthetic
+fun String.addUserAgent(): String {
```
Comment (Elyniss) [inline foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt:51]: Should we make this internal?

File: foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                cryptoModule()
+            )
+        }
+
+        val serverUrl = "wss://relay.walletconnect.com?projectId=${"test project id"}"
+        val jwt = koinApp.koin.get<JwtRepository>().generateJWT(serverUrl)
+        koinApp.modules(scarletModule(serverUrl.addUserAgent(), jwt))
+
+        val clientA = koinApp.koin.get<RelayInterface>()
+        val clientB = koinApp.koin.get<RelayInterface>()
```
Comment (Elyniss) [inline foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt:28]: Both of them will use the same jwt for authorisation right? I don't think that it should work like that.

Comment (Elyniss) [reply foundation/src/test/kotlin/com/walletconnect/foundation/RelayTest.kt:28]: Also, this makes me think if our current instrumented integration tests have the same issue. Despite that, the test works. Interesting.

File: signSDK/sign/build.gradle.kts [hunk 4]
```diff
@@ -82,6 +82,7 @@ dependencies {
 
     androidXTest()
     jUnit5()
+    jUnit5()
```
Comment (Elyniss) [inline signSDK/sign/build.gradle.kts:85]: why two jUnit5()

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 5]
```diff
@@ -105,12 +105,18 @@ fun DependencyHandlerScope.multibaseJava() {
     "api"("com.github.multiformats:java-multibase:$multibaseVersion") //https://mvnrepository.com/artifact/com.github.multiformats/java-multibase/1.1.0 vulnerability detected with library
 }
 
+fun DependencyHandlerScope.restEasyJava() {
+    "implementation"("org.jboss.resteasy:resteasy-jaxrs:$restEasyVersion")
```
Comment (jakobuid) [inline buildSrc/src/main/kotlin/Dependencies.kt:109]: Wha's that?

**Review by jakobuid — APPROVED** [review]

**Review by Elyniss — APPROVED** [review]

**Review by jakobuid — APPROVED** [review]

File: foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt (hunk 2 above)
Comment (TalhaAli00) [reply foundation/src/main/kotlin/com/walletconnect/foundation/util/UtilFunctions.kt:51]: done

File: signSDK/sign/build.gradle.kts (hunk 4 above)
Comment (TalhaAli00) [reply signSDK/sign/build.gradle.kts:85]: fixed

**Pull request merged**